src/main.cpp

```diff
@@ -4042,6 +4054,26 @@ bool ContextualCheckBlock(const CBlock &block, CValidationState &state, CBlockIn
         }
     }
 
+    // BUIP055 enforce that the fork block is > 1MB
+    // (note subsequent blocks can be <= 1MB...)
+    if (pindexPrev && pindexPrev->forkAtNextBlock(miningForkTime.value))
+    {
+        DbgAssert(block.nBlockSize, );
+        if (block.nBlockSize <= BLOCKSTREAM_CORE_MAX_BLOCK_SIZE)
+        {
+            uint256 hash = block.GetHash();
+            return state.DoS(100,
+                error("%s: BUIP055 fork block (%s, height %d) must exceed %d, but this block is %d bytes", __func__,
+                                 hash.ToString(), nHeight, BLOCKSTREAM_CORE_MAX_BLOCK_SIZE, block.nBlockSize),
+                REJECT_INVALID, "bad-blk-too-small");
+        }
+    }
+    // BUIP055 check soft-fork items, such as tx targeted to the 1MB chain
+    if (pindexPrev && pindexPrev->IsforkActiveOnNextBlock(miningForkTime.value))
+    {
+        return ValidateBUIP055Block(block, state);
```

Comment:
Could this be done inside the existing tx loop in ContextualCheckBlock, rather than a new tx loop?